### PR TITLE
Feat: Enabling Validation during training

### PR DESCRIPTION
**Pull request by desmondc65**
# PhysicsNeMo Pull Request

## Description

- Introduces separate train.zarr and valid.zarr outputs (instead of a shared file).
- Adds configuration for pointing to the validation dataset path.
- Improves output directory structure for NPZ processing (e.g., processed npz paths).
- Enables validation support in StormCast’s RWRF training workflow.
- Adds configurable arguments for `fetch_era5.py`, `fetch_rwrf.py` and `create_data.py`

basic command example for `fetch_era5.py` :
```python
python fetch_era5.py \
  --variable t2m \
  --start-date 2019/01/01 \
  --end-date 2019/01/02 \
  --hour-step 6 \
  --output-dir cache/era5
```

basic command example for `fetch_rwrf.py` :
```python
python fetch_rwrf.py \
  --variable t2m \
  --start-date 2019/01/01 \
  --end-date 2019/01/02 \
  --hour-step 6 \
  --output-dir cache/era5
```

basic command example for `create_data.py` :
```python
python create_data.py \
  --highres-vars t2m,u10,qpepre \
  --lowres-vars t2m,u10,pptn \
  --train-start 2019/08/01 \
  --train-end   2019/08/31 \
  --val-start   2019/09/01 \
  --val-end     2019/09/07 \
  --cache-base  /workspace/physicsnemo/dev/rwrf_process/cache \
  --data-base   ../data \
  --process-invariants
```

### validation at step 50 
image
image

### validation at step 100
image
image

### validation at step 100
image
image

it can be confirmed that model gradually converges instead of overfitting at the beginning.

## Related Issue
https://github.com/running-berry/physicsnemo/issues/55

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

### CODE REVIEW

**Comment by running-berry** [comment]
@desmondc65 Tho this PR looks good to me, let's keep it a bit till the cause of crooked validation loss being found out?
